iree/compiler/Dialect/Stream/Analysis/Partitioning/ReferencePartitioning.cpp

```diff
@@ -49,6 +50,18 @@ PartitionSet partitionStreamableOpsReference(
     if (op.hasTrait<OpTrait::ConstantLike>()) {
       LLVM_DEBUG(llvm::dbgs() << "(ignoring constant)\n");
       continue;
+    } else if (!isa<IREE::Stream::StreamableOpInterface>(op)) {
+      // Not a streamable op. If it has side-effects then we force a hazard on
+      // all builders so that we don't move ops across it.
+      if (!mlir::wouldOpBeTriviallyDead(&op)) {
+        LLVM_DEBUG({
+          llvm::dbgs() << "Side-effecting op forcing flush and freeze:\n";
+          op.dump();
```

Comment:
This is going to be pretty disconnected output. Can't you just `<< op` into the dbg stream?

Reply:
it always shows up side-by-side for me?
(I didn't know you could << op, I feel like I'd tried that at some point and it didn't work, but if it does then all these passes could use cleanup as they do the same thing)